src/charm.py

```diff
@@ -107,6 +115,11 @@ def _on_start(self, event: EventBase) -> None:
             event.defer()
             return
 
+        if not self.peer_relation:
```

Comment:
what if someone wants to deploy a single unit of kafka?

Reply:
Y'know, this is a VERY good point and excellent catch, definitely a mistake.

That said, all the int-tests passed, all deploying a single-unit. Why did they work?

Reply:
Well, just ran it:

```python
 self.peer_relation=<ops.model.Relation cluster:4>
```

Reply:
I can confirm that the relation exists with a single unit. I have a similar test in my code and it works.

Reply:
neat thanks for confirming!